src/test/java/edu/cornell/kfs/coa/fixture/A21SubAccountFixture.java

```diff
@@ -0,0 +1,56 @@
+package edu.cornell.kfs.coa.fixture;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.kuali.kfs.coa.businessobject.A21IndirectCostRecoveryAccount;
+import org.kuali.kfs.coa.businessobject.A21SubAccount;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+public enum A21SubAccountFixture {
+	A21_SUB_ACCOUNT_1111111_2222222_98_2(
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_1111111_98_PERCENT_ACTIVE
+					.getA21IndirectCostRecoveryAccountChange(),
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_2222222_2_PERCENT_ACTIVE
+					.getA21IndirectCostRecoveryAccountChange()),
+
+	A21_SUB_ACCOUNT_1111111_98_INACTIVE_98_INACTIVE_3333333_100_PERCENT_ACTIVE(
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_1111111_98_PERCENT_INACTIVE
+					.getA21IndirectCostRecoveryAccountChange(),
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_1111111_98_PERCENT_INACTIVE
+					.getA21IndirectCostRecoveryAccountChange(),
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_3333333_100_PERCENT_ACTIVE
+					.getA21IndirectCostRecoveryAccountChange()),
+
+	A21_SUB_ACCOUNT_3333333_100(
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_3333333_100_PERCENT_ACTIVE
+					.getA21IndirectCostRecoveryAccountChange()),
+
+	A21_SUB_ACCOUNT_3333333_100_INACTIVATE(
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_3333333_100_PERCENT_ACTIVE
+					.getA21IndirectCostRecoveryAccountChange(),
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_1111111_98_PERCENT_INACTIVE
+					.getA21IndirectCostRecoveryAccountChange(),
+			A21IndirectCostRecoveryAccountFixture.A21_ICR_2222222_2_PERCENT_INACTIVE
+					.getA21IndirectCostRecoveryAccountChange());
+
+	public final A21IndirectCostRecoveryAccount[] icrAccounts;
```

Comment:
Just wanted to confirm if you saw my edits to my older comment regarding these. (Sorry if you didn't see those edits.) With regards to my edited older comment, declaring the array as final does not prevent the array contents from being updated. If you want to make this truly immutable, you should use an unmodifiable list or similar data structure, and I would suggest keeping the ICR accounts in their fixture form until the getA21SubAccount() method gets called. (You can still use a var-args parameter in the constructor and then have it wrap the values in a list.)

The same goes for the other fixtures that are set up similar to this one.